packages/core/test/utils/errors.test.ts

```diff
@@ -0,0 +1,40 @@
+import { it, describe } from 'node:test';
+import { strict as assert } from 'node:assert';
+
+import { type NodeId } from '../../src/index.js';
+import { NodeError, rivetErrorToString } from '../../src/utils/errors.js';
+
+describe('rivetErrorToString', () => {
+  it('should handle AggregateError', () => {
+    const nodeError = new Error('Error 2');
+    (nodeError as NodeError).node = {
+      data: undefined,
+      id: 'nodeId' as NodeId,
+      title: 'Node title',
+      type: 'type',
+      visualData: {} as any, // Unused
+    };
+
+    assert.equal(
+      rivetErrorToString(
+        new AggregateError(
+          [
+            new Error('Error 1', { cause: new Error('Root cause') }), //
+            nodeError,
+            'Error 3',
+            null,
+          ],
+          'Multiple errors',
+        ),
+      ),
+      `
+Multiple errors
```

Comment:
Would prefer snapshots for stuff like this, but I don't think there's an easy way to do this with the native package